### ISSUE TITLE
Edit Forms not working after upgrade to 1.0.1

### ISSUE THREAD

**Issue by Mangetsu** [sonata-project/SonataTranslationBundle#55]
I'm getting this error after upgrading to Sonata Translation 1.0.1 :
```
CRITICAL - Uncaught PHP Exception Twig_Error_Runtime: "An exception has been thrown during the rendering of a template ("Unable to find template "".") in "SonataAdminBundle:CRUD:base_edit_form.html.twig"."
```

here is my Composer.json :

```
"php":                                        ">=5.3.9",
    "symfony/symfony":                            "2.6.*",
    "doctrine/orm":                               "~2.2,>=2.2.3,<2.5",
    "doctrine/dbal":                              "<2.5",
    "doctrine/doctrine-bundle":                   "~1.4",
    "symfony/assetic-bundle":                     "~2.3",
    "symfony/swiftmailer-bundle":                 "~2.3",
    "symfony/monolog-bundle":                     "~2.4",
    "sensio/distribution-bundle":                 "~4.0",
    "sensio/framework-extra-bundle":              "~3.0,>=3.0.2",
    "incenteev/composer-parameter-handler":       "~2.0",

    "sonata-project/core-bundle":                 "^2.3",
    "sonata-project/admin-bundle":                ">=2.3.3",
    "sonata-project/user-bundle": 				  "~2.2",
    "sonata-project/intl-bundle":                 "^2.2",
    "sonata-project/media-bundle":                "~2.2",
    "sonata-project/doctrine-orm-admin-bundle":   ">=2.3.3",
    "sonata-project/translation-bundle":          "<=1.0.0",
    "sonata-project/formatter-bundle":            "^2.3",
    "pixassociates/sortable-behavior-bundle":     "0.1.*@dev",

    "stof/doctrine-extensions-bundle":            "^1.1",
    "besimple/i18n-routing-bundle":               "^2.3",

    "friendsofsymfony/user-bundle":               "~1.3",

    "doctrine/data-fixtures":                     "~1.0",
    "doctrine/doctrine-cache-bundle":             "~1.0",
    "doctrine/migrations":                        "dev-master",
    "doctrine/doctrine-migrations-bundle":        "dev-master",
    "sonata-project/cache-bundle":                "~2.1",

    "eko/feedbundle":                             "~1.0",
    "lunetics/locale-bundle":                     "~2.3",
    "liip/imagine-bundle":                        "~1.0",
    "wjzijderveld/check-bundles":                 "~1.1",
    "burgov/key-value-form-bundle":               "~1.0",

    "sonata-project/page-bundle": "~2.3",
    "sonata-project/datagrid-bundle": "2.2.*@dev",
    "sonata-project/notification-bundle": "~2.3",
    "videlalvaro/php-amqplib": "~2.5",
    "liip/monitor-bundle": "~2.1",
    "twig/extensions": "^1.3"
```

any ideas please ?

**Comment by Soullivaneuh** [comment]
Could you please past result of `composer info -i | grep sonata-project`?

**Comment by Mangetsu** [comment]
here's my installed versions

sonata-project/admin-bundle              2.3.6
sonata-project/block-bundle              2.2.13
sonata-project/cache                     1.0.6
sonata-project/cache-bundle              2.2.4
sonata-project/core-bundle               2.3.9
sonata-project/datagrid-bundle           dev-master cc42f28
sonata-project/doctrine-extensions       1.0.2
sonata-project/doctrine-orm-admin-bundle 2.3.3
sonata-project/easy-extends-bundle       2.1.9
sonata-project/exporter                  1.4.0
sonata-project/formatter-bundle          2.3.3
sonata-project/google-authenticator      1.0.2
sonata-project/intl-bundle               2.2.3
sonata-project/media-bundle              2.3.3
sonata-project/notification-bundle       2.3.2
sonata-project/page-bundle               2.3.9
sonata-project/seo-bundle                1.1.10
sonata-project/translation-bundle        1.0.0
sonata-project/user-bundle               2.2.4

**Comment by Soullivaneuh** [comment]
@nicolas-bastien @rande any idea?

**Comment by Mangetsu** [comment]
Please not that after undoing this : https://github.com/sonata-project/SonataTranslationBundle/commit/8bbcc4e36e68860e924b2496b5c9be9d7b415f20

everything is back to normal, maybe i'm using and old Admin version ?

**Comment by Soullivaneuh** [comment]
Maybe. Try to upgrade sonata-project/block-bundle.

**Comment by Skuzy** [comment]
I have same error after <code>composer update</code>
So I decided to go back to translation-bundle 1.0.0

sonata-project/admin-bundle              2.3.6   Symfony SonataAdminBundle
sonata-project/block-bundle              2.2.13  Symfony SonataBlockBundle
sonata-project/cache                     1.0.6   Cache library
sonata-project/core-bundle               2.3.9   Symfony SonataCoreBundle
sonata-project/doctrine-extensions       1.0.2   Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle 2.3.3   Symfony Sonata / Integrate Doctrine ORM into the SonataAdminBundle
sonata-project/easy-extends-bundle       2.1.9   Symfony SonataEasyExtendsBundle
sonata-project/exporter                  1.4.0   Lightweight Exporter library
sonata-project/google-authenticator      1.0.2   Library to integrate Google Authenticator into a PHP project
sonata-project/translation-bundle        1.0.0   SonataTranslationBundle
sonata-project/user-bundle               2.2.4   Symfony SonataUserBundle

And if I update my block-bundle i have this problem:

sonata-project/admin-bundle 2.3.6 requires sonata-project/block-bundle ~2.2,>=2.2.7,<2.3 -> no matching package found.

Did you fix this problem?

**Comment by Th3Mouk** [comment]
I've got the same problem while upgrading 1.0.1.

My block bundle constraint is due to the page/bundle (2.3.9) : sonata-project/block-bundle: ~2.2,<2.3

**Comment by jeancasman** [comment]
I'm getting the same issue after upgrading to Sonata Translation 1.0.1.
If I go back to Sonata Translation 1.0.0, everything works fine.  

Installed versions:

sonata-project/admin-bundle: 2.3.6
sonata-project/block-bundle: 2.2.13
sonata-project/cache: 1.0.6
sonata-project/core-bundle: 2.3.9
sonata-project/doctrine-orm-admin-bundle: 2.3.3
sonata-project/exporter: 1.4.0
sonata-project/intl-bundle: 2.2.3
sonata-project/translation-bundle: 1.0.1

**Comment by jorrit** [comment]
It seems Sonata Translation was changed to support Blocks 2.3 while support for Block 2.2 was dropped. However, Sonata Admin only supports Block 2.2. As Sonata Translation is purely an extension of Sonata Admin, I think it should support whatever version of Blocks is supported by Admin instead of the latest version.
Or am I missing something here?

**Comment by nicolas-bastien** [comment]
Hi, I have to do a fresh install this week.
I'll check that.

**Comment by ibasaw** [comment]
back to 1.0.0 'fix' the problem...

**Comment by nicolas-bastien** [comment]
@Mangetsu @ibasaw @jorrit @jeancasman @Skuzy @Soullivaneuh  first sorry to be so long to checkout your problem.

I have to admit that versionning is hard to maintain so as this bundle is pretty simple I think the best way to move on is to fit to last sonata version and let people using older version to override this service.

You can do that easily with a compiler pass, look at http://symfony.com/doc/current/cookbook/bundles/override.html

-> OverrideServiceCompilerPass
and override LocaleSwitcherBlockService

Note : don't use parameters as we gonna remove it to fit sf best practice.

I think I can add a page : "troubleshooting" to explain that in the doc.

What do your think about that ?

**Comment by sagikazarmark** [comment]
:+1: This is actually the symfony solution I think.

**Comment by Mangetsu** [comment]
i think that's the best approche indeed, thank you.

2016-01-04 15:19 GMT+00:00 Nicolas Bastien <notifications@github.com>:

> @Mangetsu <https://github.com/Mangetsu> @ibasaw
> <https://github.com/ibasaw> @jorrit <https://github.com/jorrit>
> @jeancasman <https://github.com/jeancasman> @Skuzy
> <https://github.com/Skuzy> @Soullivaneuh <https://github.com/Soullivaneuh>
> first sorry to be so long to checkout your problem.
>
> I have to admit that versionning is hard to maintain so as this bundle is
> pretty simple I think the best way to move on is to fit to last sonata
> version and let people using older version to override this service.
>
> You can do that easily with a compiler pass, look at
> http://symfony.com/doc/current/cookbook/bundles/override.html
>
> -> OverrideServiceCompilerPass
> and override LocaleSwitcherBlockService
>
> Note : don't use parameters as we gonna remove it to fit sf best practice.
>
> I think I can add a page : "troubleshooting" to explain that in the doc.
>
> What do your think about that ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sonata-project/SonataTranslationBundle/issues/55#issuecomment-168703969>
> .
>



-- 


*"Au nom de Dieu, le Tout Miséricordieux, le Très
Miséricordieux."Cordialement Badr HAKKARI*
Profil Facebook <http://fb.com/Fanopsis>
Profil Google+ <https://plus.google.com/101047329097432605161>
Profil LinkedIn <http://linkedin.com/in/hakkari>

*Tél : 0633275580*

**Comment by Mangetsu** [comment]
@nicolas-bastien you're right, i think that's the best way indeed, thank you

**Comment by WouterJ** [comment]
The problem here is that the `setDefaultOptions` method was removed from this PR in favor of `configureOptions`. Now, blockbundle 2.2 won't call the `configureOptions` method and options are not configured (resulting in an empty template).

A simple solution is to readd the `setDefaultOptions` to the block service in this bundle and call `configureOptions` in it.

Even more important is that such a big breaking thing should never make it into a *patch* version (`1.0.1`). Being strict, such things should be merged into a major version (`2.0`) (a more Sonata way is to include this in `1.1` btw).
